engine/src/main/java/org/terasology/engine/rendering/AABBRenderer.java

```diff
@@ -178,25 +178,22 @@ public Vector2fc map(int vertexIndex, float u, float v) {
     private void generateDisplayListWire() {
         float offset = 0.001f;
 
-        StandardMeshData meshData = new StandardMeshData(MeshData.DrawingMode.LINES);
+        StandardMeshData meshData = new StandardMeshData(DrawingMode.LINES);
 
 
         Vector3f dimensions = aabb.extent(new Vector3f());
-        meshData.vertices.addAll(new float[]{
-                // TOP
-                -dimensions.x - offset, -dimensions.y - offset, -dimensions.z - offset, // 0
-                +dimensions.x + offset, -dimensions.y - offset, -dimensions.z - offset, // 1
-                +dimensions.x + offset, -dimensions.y - offset, +dimensions.z + offset, // 2
-                -dimensions.x - offset, -dimensions.y - offset, +dimensions.z + offset, // 3
-
-                // BOTTOM
-                -dimensions.x - offset, +dimensions.y + offset, -dimensions.z - offset, // 4
-                +dimensions.x + offset, +dimensions.y + offset, -dimensions.z - offset, // 5
-                +dimensions.x + offset, +dimensions.y + offset, +dimensions.z + offset, // 6
-                -dimensions.x - offset, +dimensions.y + offset, +dimensions.z + offset  // 7
-        });
-
-        meshData.getIndices().addAll(new int[]{
+        Vector3f pos = new Vector3f();
+        meshData.position.put(pos.set(-dimensions.x - offset, -dimensions.y - offset, -dimensions.z - offset)); // 0
+        meshData.position.put(pos.set(+dimensions.x + offset, -dimensions.y - offset, -dimensions.z - offset)); // 1
+        meshData.position.put(pos.set(+dimensions.x + offset, -dimensions.y - offset, +dimensions.z + offset)); // 2
+        meshData.position.put(pos.set(-dimensions.x - offset, -dimensions.y - offset, +dimensions.z + offset)); // 3
+
+        meshData.position.put(pos.set(-dimensions.x - offset, +dimensions.y + offset, -dimensions.z - offset)); // 4
+        meshData.position.put(pos.set(+dimensions.x + offset, +dimensions.y + offset, -dimensions.z - offset)); // 5
+        meshData.position.put(pos.set(+dimensions.x + offset, +dimensions.y + offset, +dimensions.z + offset)); // 6
+        meshData.position.put(pos.set(-dimensions.x - offset, +dimensions.y + offset, +dimensions.z + offset)); // 7
```

Comment:
this passes in the same vector object with changed values each time. with a quick search, it was hard for me to see whether we store this element directly and would override previous states, or whether we do a "copy-on-write" somewhere. It would help a lot if we could find a way to document this somewhere (here, on the `put` method of the `VertexAttribute`, I think).